content/en/docs/guides/helmreleases.md

```diff
@@ -64,6 +67,81 @@ Helm repositories. See the [`HelmRepository` CRD docs](../components/source/helm
 for more details.
 {{% /alert %}}
 
+#### Helm OCI repository
+
+The source-controller performs the Helm repository url validation i.e. the url is
+a valid OCI registry url.
+
+The URL is expected to point to a registry repository and to start with `oci://`.
+
+```yaml
+---
+apiVersion: source.toolkit.fluxcd.io/v1beta2
+kind: HelmRepository
+metadata:
+  name: podinfo
+  namespace: default
+spec:
+  type: oci
+  interval: 5m0s
+  url: oci://ghcr.io/stefanprodan/charts
+```
+
+#### Helm repository authentication with credentials
+
+In order to use a private Helm repository, you may need to provide the credentials.
+
+For HTTP/S repositories, the credentials can be provided as a secret reference with
+basic authentication.
+
+```yaml
+apiVersion: source.toolkit.fluxcd.io/v1beta2
+kind: HelmRepository
+metadata:
+  name: podinfo
+  namespace: default
+spec:
+  interval: 1m
+  url: https://stefanprodan.github.io/podinfo
+  secretRef:
+    name: example-user
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: example-user
+  namespace: default
+stringData:
+  username: example
+  password: 123456
+```
+
+For OCI repositories, the credentials can be provided alternatively as a secret reference
+with dockerconfig authentication.
+
+```yaml
+---
+apiVersion: source.toolkit.fluxcd.io/v1beta2
+kind: HelmRepository
+metadata:
+  name: podinfo
+  namespace: default
+spec:
+  interval: 5m0s
+  url: oci://ghcr.io/stefanprodan/charts
+  type: "oci"
+  secretRef:
+    name: regcred
+```
+
+regcred can be created with kubectl:
+
+```shell
+kubectl create secret generic regcred \
+    --from-file=.dockerconfigjson=<path/to/.docker/config.json> \
```

Comment:
This will not work as in most cases the config uses credential helpers. Please use:

```
kubectl create secret docker-registry regcred \
 --docker-server=ghcr.io \
 --docker-username=gh-user \
 --docker-password=gh-token
```